communication/src/AsyncTlsConnection.cpp

```diff
@@ -128,7 +128,7 @@ void AsyncTlsConnection::readMsg() {
         histograms_.received_msg_size->recordAtomic(bytes_transferred);
         {
           concord::diagnostics::TimeRecorder<true> scoped_timer(*histograms_.read_enqueue_time);
-          receiver_->onNewMessage(peer_id_.value(), read_msg_.data(), bytes_transferred);
+          receiver_->onNewMessage(peer_id_.value(), read_msg_.data(), bytes_transferred, peer_id_.value());
```

Comment:
The last parameter should be taken from the message header.

Reply:
done.